### ISSUE TITLE
[Digitizer] Feature Attributes UI is missing default values which are configured in Layer Properties->Attributes Form

### ISSUE THREAD

**Issue by AlexanderFinkbeiner** [NationalSecurityAgency/qgis-latlontools-plugin#52]
Reproduction steps:
1. Add a default value at Layer properties->Attributes Form->Defaults 
![image](https://user-images.githubusercontent.com/22540563/97423104-0b511880-190f-11eb-945f-0239926bb5fb.png)
2. Use "Lat Long Digitize" to add a feature and enter the coordinates 
3. At the attribute form the default value is not set: 
![image](https://user-images.githubusercontent.com/22540563/97423509-a0eca800-190f-11eb-8a99-110635d8adfa.png)

If I add a feature with the standard "Add Point feature" it's setting the default value: 
![image](https://user-images.githubusercontent.com/22540563/97423720-f4f78c80-190f-11eb-8b49-a14a98820957.png)

What is the difference here? I thought these two ways are calling the same method `openFeatureForm()`?

**Comment by hamiltoncj** [comment]
That is a great question which I don't have an answer to at the moment. Apparently there must be something more to it than just calling openFeatureForm. I will put in a little bit of time to see if I can find an answer.

**Comment by AlexanderFinkbeiner** [comment]
Thanks, I will also have a second look at the QGIS repo, maybe I can find the difference.

**Comment by AlexanderFinkbeiner** [comment]
I'm quite new to QGIS but I guess this will help us?! https://github.com/qgis/QGIS/blob/b5178d98537613a5110366e9bcae49d96e9cd9ad/src/app/qgsfeatureaction.cpp#L168

**Comment by SuKaltenbach** [comment]
I have the same issue at my project. Im really new to programming but is there a way to help fix this?

**Comment by hamiltoncj** [comment]
I have not found the answer to this question yet. I just posted the question to the QGIS developers group. Hopefully, someone will respond.

**Comment by rduivenvoorde** [comment]
@hamiltoncj I'm trying to debug this, and also tried to use 
```
feat = QgsFeature()            
ffeat.setFields(layer.fields())
```
instead of
```
feat = QgsFeature(layer.fields())
```
but my debugging skills and environment are a little sketchy on the moment. 
I would create a tiny dataset, and create a QGIS issue, because I think that is what it is.... (while debugging it looked like I passes some sip-exception, but I could not tell from it what went wrong, probably a coredev with better debug tools/skills can).

Mmm, look: having created a geopackage with ONLY a text (uuid) field, creates also a fid field (to be filled autogenerated...
Then also the fid field fails.
Normal digitize

![Screenshot-20201121105246-472x178](https://user-images.githubusercontent.com/731673/99874388-f4d56e80-2be7-11eb-97c0-295cfb8bdcb5.png)

vs lat lon one:

![Screenshot-20201121105501-472x178](https://user-images.githubusercontent.com/731673/99874400-06b71180-2be8-11eb-8840-faa46c5ed77a.png)

It looks like the creation of the QgsFeature misses some essential step...

**Comment by AlexanderFinkbeiner** [comment]
Thanks for having a look at it! Then let's hope a core developer could shine some light on this.

**Comment by rduivenvoorde** [comment]
See https://github.com/qgis/QGIS/issues/40213 
This works:

```
            from qgis.core import QgsVectorLayerUtils
            geometry = QgsGeometry.fromPointXY(QgsPointXY(x, y))
            feat = QgsVectorLayerUtils.createFeature(layer, geometry, {}, layer.createExpressionContext() )
``

**Comment by hamiltoncj** [comment]
@rduivenvoorde Thank you so much for finding the solution. I had no idea that there was a separate feature creation function. I released version 3.4.6 that has addressed this request. It seems to work for me. Let me know if it satisfies this issue.

**Comment by AlexanderFinkbeiner** [comment]
Yes now it is working! Thanks for updating it so quickly!
